### PR TITLE
DateBox - Should not render two dropdown icons in FF on mobile device(T1197922)

### PR DESCRIPTION
**Pull request by alexanderPolosatov**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js [hunk 1]
```diff
@@ -6418,3 +6419,28 @@ QUnit.module('validation', {
...
+        const $dateBox = $('#dateBox').dxDateBox({
+            pickerType: 'native',
+            value: dateValue
+        });
+
+        const $dropDownButton = $dateBox.find(`.${DROP_DOWN_BUTTON_CLASS}`);
+
+        assert.strictEqual($dropDownButton.length, isMozilla ? 0 : 1, 'dropDownButton');
+    });
+}
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js]: mis indent
Let's move it to the next line

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js [hunk 2]
```diff
@@ -6418,3 +6419,28 @@ QUnit.module('validation', {
         assert.strictEqual(this.$dateBox.hasClass(SHOW_INVALID_BADGE_CLASS), false, 'validation icon is be hidden');
     });
 });
+
+QUnit.module('Device specific tests', {
+    beforeEach: function() {
+        this._savedDevice = devices.current();
+        devices.real({ platform: 'android', deviceType: 'phone', version: [4, 4, 2], android: true });
+        devices.current({ platform: 'android' });
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js]: do we really need `current` here?

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js [hunk 3]
```diff
@@ -6418,3 +6419,28 @@ QUnit.module('validation', {
...
+        const dateValue = new Date(2016, 6, 15, 14, 30);
+        const isMozilla = browser.mozilla;
+        const $dateBox = $('#dateBox').dxDateBox({
+            pickerType: 'native',
+            value: dateValue
+        });
+
+        const $dropDownButton = $dateBox.find(`.${DROP_DOWN_BUTTON_CLASS}`);
+
+        assert.strictEqual($dropDownButton.length, isMozilla ? 0 : 1, 'dropDownButton');
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js]:
```suggestion
        assert.strictEqual($dropDownButton.length, isMozilla ? 0 : 1, `dropDownButton is ${isMozilla ? 'not' : ''} rendered`);
```

File: packages/devextreme/js/ui/date_box/ui.date_box.base.js [hunk 4]
```diff
@@ -151,7 +152,9 @@ const DateBox = DropDownEditor.inherit({
 
     _updatePickerOptions: function() {
         let pickerType = this.option('pickerType');
+        const platform = devices.real().platform;
         const type = this.option('type');
+        let showDropDownButtonValue;
```
Comment (ksercs) [inline packages/devextreme/js/ui/date_box/ui.date_box.base.js]:
```suggestion
        let showDropDownButtonValue;
```
```suggestion
        let showDropDownButton;
```

File: packages/devextreme/js/ui/date_box/ui.date_box.base.js [hunk 5]
```diff
@@ -161,7 +164,16 @@ const DateBox = DropDownEditor.inherit({
...
-        this.option('showDropDownButton', devices.real().platform !== 'generic' || pickerType !== PICKER_TYPE['native']);
+        const isMozillaOnAndroid = platform === 'android' && browser.mozilla;
+        const isNativePickerType = pickerType === PICKER_TYPE.native;
+        if(isNativePickerType && isMozillaOnAndroid) {
+            showDropDownButtonValue = false;
+        } else {
+            showDropDownButtonValue = platform !== 'generic' || !isNativePickerType;
+        }
+
+        this.option('showDropDownButton', showDropDownButtonValue);
```
Comment (ksercs) [inline packages/devextreme/js/ui/date_box/ui.date_box.base.js]:
```suggestion
        this.option({ showDropDownButton });
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js (hunk 2 above)
Comment (alexanderPolosatov) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js]: fixed

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js (hunk 1 above)
Comment (alexanderPolosatov) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js]: fixed

File: packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js (hunk 3 above)
Comment (alexanderPolosatov) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js]: fixed

File: packages/devextreme/js/ui/date_box/ui.date_box.base.js (hunk 5 above)
Comment (alexanderPolosatov) [reply packages/devextreme/js/ui/date_box/ui.date_box.base.js]: done

File: packages/devextreme/js/ui/date_box/ui.date_box.base.js (hunk 4 above)
Comment (alexanderPolosatov) [reply packages/devextreme/js/ui/date_box/ui.date_box.base.js]: done

File: packages/devextreme/js/ui/date_box/ui.date_box.base.js [hunk 6]
```diff
@@ -161,7 +164,16 @@ const DateBox = DropDownEditor.inherit({
             pickerType = PICKER_TYPE.list;
         }
 
-        this.option('showDropDownButton', devices.real().platform !== 'generic' || pickerType !== PICKER_TYPE['native']);
+        const isMozillaOnAndroid = platform === 'android' && browser.mozilla;
+        const isNativePickerType = pickerType === PICKER_TYPE.native;
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/date_box/ui.date_box.base.js]: Let's extract this logic to separate function.

Comment (EugeniyKiyashko) [reply packages/devextreme/js/ui/date_box/ui.date_box.base.js]: Also let's use _isNativeType() method instead

File: packages/devextreme/js/ui/date_box/ui.date_box.base.js [hunk 7]
```diff
@@ -161,7 +164,16 @@ const DateBox = DropDownEditor.inherit({
...
+        if(isNativePickerType && isMozillaOnAndroid) {
```
Comment (EugeniyKiyashko) [inline packages/devextreme/js/ui/date_box/ui.date_box.base.js]: May be we can simplify this condition?

**Review by EugeniyKiyashko — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

File: packages/devextreme/js/ui/date_box/ui.date_box.base.js (hunk 7 above)
Comment (alexanderPolosatov) [reply packages/devextreme/js/ui/date_box/ui.date_box.base.js]: fixed

File: packages/devextreme/js/ui/date_box/ui.date_box.base.js (hunk 6 above)
Comment (alexanderPolosatov) [reply packages/devextreme/js/ui/date_box/ui.date_box.base.js]: fixed

**Pull request merged**